frontend/packages/console-shared/src/components/drawer/Drawer.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { DraggableData, DraggableCore } from 'react-draggable';
+import { useSize } from './drawer-utils';
+import './Drawer.scss';
+
+type DrawerProps = {
+  height?: number;
+  defaultHeight?: number;
+  open?: boolean;
+  defaultOpen?: boolean;
+  maxHeight?: number | string;
+  isResizable?: boolean;
+  header: React.ReactNode;
+  headerClassName?: string;
+  onChange?: (open: boolean, height: number) => void;
+};
+
+const Drawer: React.FC<DrawerProps> = ({
+  children,
+  defaultHeight,
+  height,
+  maxHeight = '100%',
+  defaultOpen,
+  open = defaultOpen,
+  isResizable = false,
+  header,
+  headerClassName,
+  onChange,
+}) => {
+  const [heightState, setHeightState] = React.useState(defaultHeight);
+  const [lastObservedHeight, setLastObservedHeight] = React.useState<number>(null);
+  const [size, ref] = useSize<HTMLDivElement>();
+  const resizeHeight = height || heightState;
+  const minimumHeight = size.height || 0;
+  const handleDrag = (e: MouseEvent, data: DraggableData) => {
+    const newHeight = resizeHeight - data.y;
+    if (newHeight < 0) return;
+    if (onChange) {
+      onChange(open, newHeight);
+    } else {
+      setHeightState(newHeight);
+    }
+  };
+
+  const handleResizeStart = (e: MouseEvent, data: DraggableData) => {
+    const newHeight = resizeHeight - data.y;
+    if (newHeight >= minimumHeight) onChange(true, newHeight);
```

Comment:
Firstly, why are you notifying changes at start time. There shouldn't be a delta.
Secondly, onChange can be undefined.

Reply:
There is fluctuation while dragging the header from a minimized state. will be removing this.